pkg/cache/crcache/repository.go

```diff
@@ -59,6 +63,7 @@ type cachedRepository struct {
 
 	metadataStore        meta.MetadataStore
 	repoPRChangeNotifier cachetypes.RepoPRChangeNotifier
+	coreClient           client.WithWatch
```

Comment:
Is it a good project/dependency structure, that the cache repository needs to reach out to the k8s layer directly? Not sure, just asking if this should be the way to go.